R-package/R/osf-interactions.R

```diff
@@ -0,0 +1,174 @@
+    write_dir <- ""
+    if (current_file_info$destination == "osf") {
+      write_dir <- file.path(reused_storage_prefix,
+                             current_file_info$osf_location)
+      dir.create(write_dir,
+                 showWarnings = FALSE,
+                 recursive = TRUE)
+      
+    }
+    
+    
+    if(write_dir != ""){
+      if (current_file_info$file_type == "excel") {
+        # Use parts as names of sheets to extract
+        sheets_to_copy <- current_file_info$sheet_names
+        tmp_workbook <- openxlsx::copyWorkbook(dictionary)
+        # Loop over sheets removing unnecessary sheets
+        existing_sheets <- names(tmp_workbook)
+        for (sheet_name in existing_sheets) {
+          if (!(sheet_name %in% sheets_to_copy)) {
+            openxlsx::removeWorksheet(tmp_workbook, sheet_name)
+          }
+        }
+        
+        # Save the workbook in the appropriate directory
+        openxlsx::saveWorkbook(tmp_workbook,
+                               file = file.path(
+                                 write_dir,
+                                 paste0(current_file_info$file_name, ".xlsx")
+                               ),
+                               overwrite = TRUE)
+        
+      } else if (current_file_info$file_type == "csv") {
+        sheet_name <- current_file_info$sheet_names
+        output_sheet <-
+          openxlsx::readWorkbook(dictionary, sheet_name)
+        if (current_file_info$add_headers != odm_dictionary$dictionary_missing_value) {
+          new_headers <- strsplit(current_file_info$add_headers, ";")[[1]]
+          output_sheet <-
+            rbind(colnames(output_sheet), output_sheet)
+          if (length(colnames(output_sheet)) > length(new_headers)) {
+            length_to_append <-
+              length(colnames(output_sheet)) - length(new_headers)
+            new_headers <- c(new_headers, rep("", length_to_append))
+          } else if (length(colnames(output_sheet)) < length(new_headers)) {
+            length_to_append <-
+              length(new_headers) - length(colnames(output_sheet))
+            for (col_counter in 1:length_to_append) {
+              output_sheet <- cbind(output_sheet, "")
+            }
+          }
+          colnames(output_sheet) <- new_headers
+        }
+        
+        write.csv(output_sheet,
+                  file = file.path(write_dir,
+                                   paste0(
+                                     current_file_info$file_name, ".csv"
+                                   )),
+                  row.names = FALSE)
+      }
+    }
+  }
+}
+
+
+#' Archive previous release
+#' 
+#' This function archives the previous release on OSF. It creates a new archive folder, moves the previous release to the archive folder.
+#' 
+#' @param OSF_TOKEN The OSF token used to authenticate the user.
+#' @param OSF_LINK The OSF link to the main repo.
+archive_previous_release <- function(OSF_TOKEN, OSF_LINK, past_dictionary_version){
```

Comment:
Should be split into functions. Once which creates the archive and one which uploads the new release.